src/Raven.Server/Documents/Patch/JavaScriptUtils.cs

```diff
@@ -48,6 +51,73 @@ public JavaScriptUtils(ScriptRunner runner, Engine engine)
             _scriptEngine = engine;
         }
 
+        internal JsValue Count(JsValue self, JsValue[] args)
+        {
+            if (args.Length != 0 || (CurrentlyProcessedObject is not BlittableObjectInstance doc)) 
+            {
+                throw new InvalidOperationException("count() must be called without arguments");
+            }
+            
+            doc.TryGetValue(Constants.Documents.Indexing.Fields.CountFieldName, out var countValue);
+            
+            return countValue;
+        }
+        
+        internal JsValue Key(JsValue self, JsValue[] args)
+        {
+            if (args.Length != 0 || (CurrentlyProcessedObject is not BlittableObjectInstance doc)) 
+            {
+                throw new InvalidOperationException("key() must be called without arguments");
+            }
+            
+            var groupByFields = doc.Projection._query.Metadata.GroupBy;
+
+            if (groupByFields.Length == 1)
+            {
+                doc.TryGetValue(groupByFields[0].Name.Value, out var keyValue);
+
+                return keyValue;
+            }
+
+            else
+            {
+                var res = new DynamicJsonValue();
+
+                foreach (var field in groupByFields)
+                {
+                    if (doc.Blittable.TryGetMember(field.Name.Value, out var keyValue))
+                        res.Properties.Add((field.Name.Value, keyValue));
+                }
+
+                var jsonFromCtx = Context.ReadObject(res, null);
+                JsValue resJs = TranslateToJs(_scriptEngine, Context, jsonFromCtx);
+
+                return resJs;
+            }
+        }
+        
+        internal JsValue Sum(JsValue self, JsValue[] args)
+        {
+            if (args.Length != 1 || CurrentlyProcessedObject is not BlittableObjectInstance doc)
+            {
+                throw new InvalidOperationException("sum(field) must be called with a single argument");
+            }
+
+            if (args[0] is ScriptFunctionInstance sfi)
+            {
+                if (sfi.FunctionDeclaration.ChildNodes[1] is StaticMemberExpression sme)
+                {
+                    if (sme.Property is Identifier identifier)
+                    {
+                        doc.TryGetValue(identifier.Name, out var sumValue);
+                        
+                        return sumValue;
+                    }
+                }
+            }
+            return null;
```

Comment:
JsValue.Null or JsValue.Undefined?

Reply:
Now we're throwing an exception there, since it means the argument is not correct arrow function expression, and we only want to handle such case